test/IDE/issue-65843.swift

```diff
@@ -0,0 +1,7 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-ide-test -batch-code-completion -source-filename %s -filecheck %raw-FileCheck -completion-output-dir %t
+
+protocol P {}
+
+(any P).#^COMPLETE?check=META^#
+// META: Keyword/CurrNominal:  Type[#(any P).Type#]; name=Type
```

Comment:
Please add a test for `(P)`. `P.Type` == `(P).Type`, so the completion should be an *existential* metatype. I will try to find a better place for these tests in the meantime. Issue-specific test files are more of a last resort in organizing tests.

Reply:
Sorry, I accidentally truncated my reply. I meant the test case to be `(P)`, not `P.Type`, etc.

Reply:
Ok.

Reply:
This test currently fails as expected, since the paren sugar check would work for `(any P)` and not for `(P)`.